crates/biome_graphql_parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,720 @@
+                            .with_detail(self.position..self.position + 1, "line breaks here"),
+                    ),
+                ),
+                _ => {
+                    self.advance_char_unchecked();
+                    (state, None)
+                }
+            },
+            LexStringState::InBlockString => match chr {
+                b'"' => (self.consume_quote_in_string(chr, state), None),
+                b'\\' => self.consume_escape_sequence_in_string(chr, state),
+                _ => {
+                    self.advance_char_unchecked();
+                    (state, None)
+                }
+            },
+            _ => panic!("String terminated"),
+        }
+    }
+
+    fn consume_quote_in_string(&mut self, chr: u8, state: LexStringState) -> LexStringState {
+        assert_eq!(chr, b'"');
+        self.advance(1);
+        match state {
+            LexStringState::Uninitialized => {
+                if self.current_byte() == Some(b'"') {
+                    self.advance(1);
+                    LexStringState::InBlockString
+                } else {
+                    // an empty string
+                    LexStringState::Terminated
+                }
+            }
+            LexStringState::InString => LexStringState::Terminated,
+            LexStringState::InBlockString => {
+                if self.current_byte() == Some(b'"') && self.byte_at(1) == Some(b'"') {
+                    self.advance(2);
+                    LexStringState::Terminated
+                } else {
+                    state
+                }
+            }
+            _ => panic!("String terminated"),
+        }
+    }
+
+    fn consume_escape_sequence_in_string(
+        &mut self,
+        chr: u8,
+        state: LexStringState,
+    ) -> (LexStringState, Option<ParseDiagnostic>) {
+        assert_eq!(chr, b'\\');
+        let escape_start = self.text_position();
+        self.advance(1);
+        match state {
+            // '\t' etc
+            LexStringState::InString => match self.current_byte() {
+                Some(b'"' | b'\\' | b'/' | b'b' | b'f' | b'n' | b'r' | b't') => {
+                    self.advance(1);
+                    (state, None)
+                }
+
+                Some(b'u') => match self.consume_unicode_escape() {
+                    Err(diagnostic) => (state, Some(diagnostic)),
+                    Ok(_) => (state, None),
+                },
+
+                Some(_) => {
+                    let c = self.current_char_unchecked();
+                    let diagnostic = ParseDiagnostic::new(
+                        "Invalid escape sequence",
+                        escape_start..self.text_position() + c.text_len(),
+                    ).with_hint(r#"Valid escape sequences are: `\\`, `\/`, `/"`, `\b\`, `\f`, `\n`, `\r`, `\t` or any unicode escape sequence `\uXXXX` where X is hexedecimal number. "#);
```

Comment:
Consider using `with_alternatives` to present a list of things to the user instead.